### PR TITLE
feat(components): add attachment component

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
Existing attachment in design system doesn't support multi file uploads. 

This PR creates an attachment that supports multi file uploads, which will be used subsequently for bulk uploading of modals 

## Solution
1. add attachment component with some changes from design system - exposed methods now take `File[]` and we also expose an additional prop for consuming `FileRejection[]`
2. add new component for displaying rejections. not reusing the original one because it felt abit too much to stuff it all in 1 single file.
3. rework styling for attachment 

## Screenshots
**AFTER**:
**base state, empty**
Screenshot 2023-11-01 at 2 14 35 PM

**with content + rejections**
Screenshot 2023-11-01 at 2 15 11 PM

## Tests
no tests - coming in follow up, when component is actually used kekw

### CODE REVIEW

File: src/components/Attachment/Attachment.tsx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+import type { Promisable } from "type-fest"
+
+import { AttachmentStylesProvider } from "./AttachmentContext"
+import { AttachmentDropzone } from "./AttachmentDropzone"
+import { AttachmentError } from "./AttachmentError"
+import { AttachmentFileInfo } from "./AttachmentFileInfo"
+import { MAX_NUM_UPLOADS } from "./constants"
+import { getReadableFileSize } from "./utils"
+
+export interface AttachmentProps extends UseFormControlProps<HTMLElement> {
```
Comment (dcshzj) [inline src/components/Attachment/Attachment.tsx:25]: Should we also use the `maxFiles` prop?

File: src/components/Attachment/Attachment.tsx [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            />
+          </Box>
+          {showMaxSize && (
+            <Text
+              id={maxSizeTextId}
+              color="base.content.medium"
+              mt="0.5rem"
+              textStyle="body-2"
+            >
+              Maximum size: {readableMaxSize}
```
Comment (dcshzj) [inline src/components/Attachment/Attachment.tsx:215]:
```suggestion
              Maximum size per image: {readableMaxSize}
```

File: src/theme/components/index.ts [hunk 3]
```diff
@@ -12,5 +12,5 @@ export const components = {
   Checkbox,
   Breadcrumb,
   Infobox,
-  Rating,
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply src/theme/components/index.ts:15]: the import wasn't being used?!

File: src/components/Attachment/utils.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const DECIMAL_BYTE_UNITS = ["B", "kB", "MB", "GB", "TB"]
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply src/components/Attachment/utils.ts:1]: actually no leh

File: src/components/Attachment/Attachment.tsx (hunk 1 above)
Comment (seaerchin) [reply src/components/Attachment/Attachment.tsx:25]: this got changed downstream!

File: src/components/Attachment/utils.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dcshzj) [reply src/components/Attachment/utils.ts:1]: Wah okay TIL!

File: src/theme/components/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dcshzj) [reply src/theme/components/index.ts:15]: Wah okay, should delete the Rating theme stuff also while at it.

**Comment by dcshzj** [comment]
Can update the base branch to develop + rebase le since the original PR has been merged.

**Review by dcshzj — APPROVED** [review]
Non-blocking comments left, just need to update the base branch to develop.

File: src/components/Attachment/AttachmentError.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        </VisuallyHidden>
+        {imagePreview && previewSrc && (
+          <Image
+            alt="uploaded image preview"
+            sx={styles.fileInfoImage}
+            src={previewSrc}
+          />
+        )}
+        <Flex sx={styles.fileInfo}>
+          <Icon as={BxError} />
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply src/components/Attachment/AttachmentError.tsx:90]: swapped

File: src/theme/components/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (seaerchin) [reply src/theme/components/index.ts:15]: donezxc

**Pull request merged**